### DISCUSSION TITLE
Toggling settings like `showImplicitArguments`

### DISCUSSION THREAD

**Discussion by TimWSpence** [scalameta/nvim-metals#574] [Q&A] [answered]
Apologies if this is just my inexperience with neovim but is there a way to dynamically toggle metals config like `showImplicitArguments` and `showInferredType`? I did this quite frequently via the emacs plugin so I believe it is theoretically possible but have no idea how that functionality is exposed

**Comment by ckipp01** [comment, marked as answer]
Hey @TimWSpence, yup, we have a `toggle_setting` function in the Lua API that you can use for this. You can see some notes about it here:

https://github.com/scalameta/nvim-metals/blob/1b9053139342ed9ab47b29e5a1076fcbd971a565/doc/metals.txt#L1027-L1040

**Reply by TimWSpence** [reply]
> Hey @TimWSpence, yup, we have a `toggle_setting` function in the Lua API that you can use for this. You can see some notes about it here:
> 
> https://github.com/scalameta/nvim-metals/blob/1b9053139342ed9ab47b29e5a1076fcbd971a565/doc/metals.txt#L1027-L1040

Aha! I thought it probably existed somewhere. You're a legend ❤️ Is that documented somewhere? I'm very happy to attempt to raise a PR if not - feels like very useful functionality for most people

**Reply by ckipp01** [reply]
Ah, well I mean what I linked _is_ the documentation ha. Everything for `nvim-metals` should be contained in there. So exploring `:h nvim-metals` will give you all sorts of info.

**Reply by TimWSpence** [reply]
You can tell I'm new to nvim... 😅 Although I should have guessed that from the file path. Anyway, thanks so much for your help!
